system/lib/pthread/pthread_kill.c

```diff
@@ -5,11 +5,28 @@
  * found in the LICENSE file.
  */
 
-#include <emscripten/threading.h>
+#include <threading_internal.h>
+#include <emscripten/proxying.h>
+#include <emscripten/console.h>
+#include <emscripten_internal.h>
 
 #include "pthread_impl.h"
 #include "lock.h"
 
+void do_raise(void* arg) {
+  int sig = (intptr_t)arg;
+  if (sig == SIGCANCEL) {
+    // For SIGCANCEL is simply enough to run the proxied function since we call
```

Comment:
```suggestion
    // For SIGCANCEL it is simply enough to run the proxied function since we call
```

Reply:
"the proxied function" refers to the signal handler setup by sigaction that raise will call.